via/views/api/youtube.py

```diff
@@ -1,1328 +1,42 @@
-from pyramid.httpexceptions import HTTPNotFound
+import logging
+
+from h_pyramid_sentry import report_exception as report_exception_to_sentry
 from pyramid.view import view_config
 
+from via.services import YouTubeService
+
+logger = logging.getLogger(__name__)
+
 
 @view_config(route_name="api.youtube.transcript", permission="api", renderer="json")
 def transcript(request):
     """Return the transcript of a given YouTube video."""
 
     video_id = request.matchdict["video_id"]
 
-    if video_id != "1":
-        raise HTTPNotFound()
+    try:
+        transcript_ = request.find_service(YouTubeService).get_transcript(video_id)
+    except Exception as exc:  # pylint: disable=broad-exception-caught
+        report_exception_to_sentry(exc)
+        logger.exception(exc)
+
+        request.response.status_int = 500
+
+        return {
+            "errors": [
+                {
+                    "status": request.response.status_int,
+                    "code": "failed_to_get_transcript",
+                    "title": "Failed to get transcript from YouTube",
+                    "detail": str(exc),
+                }
+            ]
+        }
```

Comment:
Yep, so we'd just add a global exception view that applies to all `/api/` requests and that sends a valid JSON:API error response, and one of the types of exception that exception view would be able to handle would be `youtube-transcript-api`'s exceptions, that way nothing in the services or views layers needs to catch these exceptions they can just let the exception view handle them.

I don't think we need a specific `SerializableError` class such that something in the services or views would have to catch exceptions from `youtube-transcript-api` and re-raise them as `SerializableError`'s. I think we just make the exception view(s) be able to handle the `youtube-transcript-api` exceptions directly (and we'd probably write that in a generic way so that it can in fact handle any exception with a `cause` and have a fallback behaviour for when there's no `cause`). Then we don't need any exception-handling code in the services or views.

It may even be worth doing that now just to get the code in the correct shape, even though we don't even know if we'll ever add a second API view. Separate PR though